rollup-interface/src/node/services/da.rs

```diff
@@ -30,17 +32,33 @@ pub trait DaService: Send + Sync + 'static {
     >;
 
     /// The error type for fallible methods.
-    type Error: fmt::Debug + Send + Sync + Display;
-
-    /// Retrieve the data for the given height, waiting for it to be
-    /// finalized if necessary. The block, once returned, must not be reverted
-    /// without a consensus violation.
-    async fn get_finalized_at(&self, height: u64) -> Result<Self::FilteredBlock, Self::Error>;
+    type Error: core::fmt::Debug + Send + Sync + core::fmt::Display;
 
     /// Fetch the block at the given height, waiting for one to be mined if necessary.
-    /// The returned block may not be final, and can be reverted without a consensus violation
+    /// The returned block may not be final, and can be reverted without a consensus violation.
+    /// Call it for the same height are allowed to return different results.
+    /// Should always returns the block at that height on the best fork.
     async fn get_block_at(&self, height: u64) -> Result<Self::FilteredBlock, Self::Error>;
 
+    /// Fetch the [`DaSpec::BlockHeader`] of the last finalized block.
+    async fn get_last_finalized_block_header(
+        &self,
+    ) -> Result<<Self::Spec as DaSpec>::BlockHeader, Self::Error>;
```

Comment:
Does this need to be an `Option<BlockHeader>`? I suppose we can consider the genesis block of all chains to be final and expect clients to return that block even if no others have been finalized - but we may want to document that?

Reply:
In mock implementation, I return "error" if chain hasn't reached finality from the genesis.

That's a good point.